src/string.rs

```diff
@@ -1,3 +1,6 @@
+//! [`EcoString`]: a clone-on-write small-string-optimized reference
```

Comment:
How about just this?
```
//! A clone-on-write, small-string-optimized alternative to [`String`][...].
```
Otherwise, it's a bit of a keyword dump. Also added "alternative" because it doesn't contain a `String`.